components/collector/tests/source_collectors/dependency_track/test_source_up_to_dateness.py

```diff
@@ -0,0 +1,47 @@
+"""Unit tests for the Dependency-Track source up-to-dateness collector."""
+
+from datetime import datetime, timedelta
+
+from dateutil.tz import tzlocal
+
+from tests.source_collectors.source_collector_test_case import SourceCollectorTestCase
+
+
+class DependencyTrackSourceVersionTest(SourceCollectorTestCase):
+    """Unit tests for the Dependency-Track source version collector."""
+
+    METRIC_ADDITION = "min"
+    METRIC_TYPE = "source_up_to_dateness"
+    SOURCE_TYPE = "dependency_track"
+
+    async def test_source_up_to_dateness(self):
+        """Test that the source up-to-datentess can be measured."""
+        now = datetime.now(tz=tzlocal()).replace(microsecond=0)
+        yesterday = now - timedelta(days=1)
+        yesterday_timestamp = yesterday.timestamp() * 1000
+        last_week = now - timedelta(days=7)
+        last_week_timestamp = last_week.timestamp() * 1000
+        projects = [
+            {"name": "Project 1", "lastBomImport": yesterday_timestamp, "uuid": "p1"},
+            {"name": "Project 2", "lastBomImport": last_week_timestamp, "uuid": "p2"},
+        ]
+        response = await self.collect(get_request_json_return_value=projects)
+        self.assert_measurement(
+            response,
+            value="7",
```

Comment:
Didn't we want to be able to distinguish between the age of the vulnerabilities db (and age of analysis, I guess) and age of sbom? With this logic any DT installation which happens to have some old project listed, would seem to not be up to date. If there is no option to exclude certain projects from this metric, the user would be forced to delete old projects from DT

Reply:
There doesn't seem to be information about the age of the vulnerabilities db and analysis available in the API.